tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -949,11 +949,23 @@ function Use-AzureRmProfile
         }
       }
 
-      # Block user if they try to import a module to the session where a different version of the same module is already imported
-      if ($null -ne (Get-Module -Name $Module | Where-Object { $_.Version -ne $version} ))
+      # If a different profile's Azure Module was imported, block user
+      $importedModules = Get-Module "Azure*"
+      foreach ($importedModule in $importedModules) 
       {
-        Write-Error "A different version of module $module is already imported in this session. Start a new PowerShell session and retry the operation."
-        return
+        $versions = $ProfileMap.$Profile.$($importedModule.Name)
+        if ($null -ne $versions)
+        {
+          # We need the latest version in that profile to be imported. If old version was imported, block user and ask to import in a new session
+          $versionEnum = $versions.GetEnumerator()
```

Comment:
I'm wondering about the performance of this, how much does this add to the load time of the modules?